packages/firestore/src/core/firestore_client.ts

```diff
@@ -398,8 +398,22 @@ export class FirestoreClient {
   ): QueryListener {
     this.verifyNotTerminated();
     const listener = new QueryListener(query, observer, options);
-    this.asyncQueue.enqueueAndForget(() => {
-      return this.eventMgr.listen(listener);
+    this.asyncQueue.enqueueAndForget(async () => {
+      try {
+        await this.eventMgr.listen(listener);
+      } catch (e) {
+        logError(LOG_TAG, `Failed to register query: ${e}`);
+        if (e.name === 'IndexedDbTransactionError') {
+          listener.onError(
+            new FirestoreError(
+              Code.UNAVAILABLE,
+              `Failed to register query: ${e}`
```

Comment:
A few thoughts on this:

  * From a user's point of view they listen to queries (or get them) but we don't have a public concept of "registration". I think it's probably fine to just say "failed to listen to query", "failed to start a listener to query" or maybe something more generic like "failed to start query".
  * We should include details of the query that failed (a toString or similar). Whenever people are filing issues about broken queries our first instructions are to have people go figure out which query is causing the problem and they're usually at a loss. We should just include that in the error message.

Reply:
Updated to use "initialize" and also included the Query in the message.